src/main/service/publicationService.ts

```diff
@@ -122,44 +122,44 @@ export class PublicationService {
   }
 
   private findAndManipulatePartyInformation(hearing: any): void {
-    let applicant;
-    let respondent;
+    let applicant = '';
+    let respondent = '';
 
     if(hearing?.party) {
       hearing.party.forEach(party => {
 
         switch(PublicationService.convertPartyRole(party.partyRole)) {
           case 'APPLICANT_PETITIONER':
           {
-            applicant = this.createIndividualDetails(party.individualDetails);
+            applicant = this.createIndividualDetails(party.individualDetails).trim();
             applicant += this.stringDelimiter(applicant?.length, ',');
             break;
           }
           case 'APPLICANT_PETITIONER_REPRESENTATIVE':
           {
-            applicant += party?.friendlyRoleName;
-            applicant += this.stringDelimiter(party?.friendlyRoleName.length, ':');
-            applicant += this.createIndividualDetails(party.individualDetails);
+            applicant += this.writeStringIfValid(party?.friendlyRoleName);
+            applicant += this.stringDelimiter(this.writeStringIfValid(party?.friendlyRoleName).length, ':');
+            applicant += this.createIndividualDetails(party.individualDetails).trim();
             break;
           }
           case 'RESPONDENT':
           {
-            respondent = this.createIndividualDetails(party.individualDetails);
+            respondent = this.createIndividualDetails(party.individualDetails).trim();
             respondent += this.stringDelimiter(respondent?.length, ',');
             break;
           }
           case 'RESPONDENT_REPRESENTATIVE':
           {
-            respondent += party?.friendlyRoleName;
-            respondent += this.stringDelimiter(party?.friendlyRoleName.length, ':');
-            respondent += this.createIndividualDetails(party.individualDetails);
+            respondent += this.writeStringIfValid(party?.friendlyRoleName);
+            respondent += this.stringDelimiter(this.writeStringIfValid(party?.friendlyRoleName).length, ':');
```

Comment:
Same as above re friendlyRoleName

Reply:
Yes, you are right. I hardcode the LEGALADVISOR in code, but it will be replaced with English or Welsh in nunjucks. That's how we are doing for some other text as well.